pym/portage/dbapi/bintree.py

```diff
@@ -373,6 +382,17 @@ def __init__(self, _unused=DeprecationWarning, pkgdir=None,
 				self._pkgindex_inherited_keys,
 				chain(*self._pkgindex_translated_keys)
 			))
+			self.suffix = _suffix
+			try:
+				binpkg_compression = settings.get("BINPKG_COMPRESSION", "bzip2")
+				self.compression = _compressors[binpkg_compression]
+			except KeyError as e:
+				raise ValueError("Invalid or unsupported compression method: %s" % e.args[0])
+
+			compression_binary = varexpand(shlex_split(self.compression["compress"])[0])
+			if find_binary(compression_binary) is None:
+				missing_package = self.compression["package"]
+				raise ValueError("File compression unsupported %s. Missing package: %s" % (binpkg_compression, missing_package))
```

Comment:
You're not allowed to us portage.settings there. Whatever code uses it those variables should have a config instance and use portage.util.varexpand() to evaluate the variables on-demand.